File: course_01/contest_04/04/Titanic.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"strconv"
	"strings"
)

type Passenger struct {
	age int
	name string
}

func main() {
	var (
		pclass, age, a, c int
		line []string
		t []string
		v []Passenger
		p Passenger
		s, n string
	)

	fmt.Scan(&pclass, &age)

	file, err := os.Open("train.csv")
	if err != nil {
		log.Fatal(err)
	}
	defer file.Close()

	S := bufio.NewScanner(file)
	for S.Scan() {
		line = strings.Split(S.Text(), "\r")
		for _, w := range line {
			t = strings.Split(w, ",")
			a, err = strconv.Atoi(t[6])
			c, err = strconv.Atoi(t[2])
			if t[5] == "female" && a > age && c == pclass {
				s = t[3] + "," + t[4]
				n = ""
				for i := 1; i < len(s)-1; i++ {
					if !(s[i-1] == '"' && s[i] == '"') {
						n += string(s[i])
					}
				}
				p = Passenger{name: n, age: a}
				v = append(v, p)
			}
		}
	}

	for i := 0; i < len(v); i++ {
		for j := i + 1; j < len(v); j++ {
			if v[i].age > v[j].age || (v[i].age == v[j].age && v[i].name > v[j].name) {
				p = v[i]
				v[i] = v[j]
				v[j] = p
			}
		}
	}

	for _, i := range v {
		fmt.Println(i.name)
	}
}

File: course_01/contest_04/03/Titanic.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <sstream>
#include <vector>

using namespace std;

struct Passenger {
    std::string name;
    int age;
};

int main()
{
    int pclass, age;            
    string a, b, c, line;
    std::vector<std::string> t; 
    std::vector<Passenger> v;    
    Passenger p;                 
 
    std::cin >> pclass >> age;
    std::ifstream file("train.csv");

    while (not file.eof()) {
        std::getline(file, line, '\r');

        std::istringstream ss(line);
        while (std::getline(ss, a, ',')) {
            if (a.size() == 0)
                a = "0";
            t.push_back(a);
        }

        if (t[5] == "female" && std::stoi(t[6]) > age && std::stoi(t[2]) == pclass) {
            b = t[3] + ',' + t[4];
            c = "";
            for (int i = 1; i < b.size() - 1; i++)
                if (!(b[i - 1] == '\"' && b[i] == '\"'))
                    c += b[i];
            p = { c, std::stoi(t[6]) };
            v.push_back(p);
        }
        t.clear();
    }
    file.close();

    for (int i = 0; i < v.size(); i++)
        for (int j = i + 1; j < v.size(); j++)
            if (v[i].age > v[j].age || (v[i].age == v[j].age && v[i].name > v[j].name)) {
                p = v[i];
                v[i] = v[j];
                v[j] = p;
            }

    for (auto& i : v)
        std::cout << i.name << "\n";
}
